ssl/statem/extensions.c

```diff
@@ -864,11 +881,15 @@ int tls_construct_extensions(SSL_CONNECTION *s, WPACKET *pkt,
     }
 
     if (!WPACKET_close(pkt)) {
-        SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
-        return 0;
+        error = ERR_R_INTERNAL_ERROR;
+        goto err;
     }
 
     return 1;
+ err:
+    if (!for_comp && error != -1)
+        SSLfatal(s, SSL_AD_INTERNAL_ERROR, error);
```

Comment:
By moving this `SSLfatal` call we are losing information about the error, i.e. specifically the line number of the error which the SSLfatal macro automatically passes

Reply:
Gotcha.